File: service/src/main/java/no/nav/melosys/service/dokument/brev/BrevDataInnvilgelseFlereLand.java
package no.nav.melosys.service.dokument.brev;

import java.util.Collection;
import java.util.List;

import no.nav.melosys.domain.Lovvalgsperiode;
import no.nav.melosys.domain.avklartefakta.AvklartVirksomhet;
import no.nav.melosys.domain.kodeverk.Landkoder;

public final class BrevDataInnvilgelseFlereLand extends BrevData {
    public Collection<AvklartVirksomhet> arbeidsgivere;
    public List<String> alleArbeidsland;
    public Lovvalgsperiode lovvalgsperiode;
    public boolean harAvklartMaritimTypeSokkel;
    public boolean harAvklartMaritimTypeSkip;
    public String bostedsland;
    public boolean erBegrensetPeriode;
    public boolean erMarginaltArbeid;
    public Landkoder trydemyndighetsland;

    public BrevDataA1 vedleggA1;

    public BrevDataInnvilgelseFlereLand(BrevbestillingRequest brevbestillingRequest, String saksbehandler) {
        super(brevbestillingRequest, saksbehandler);
    }
}

File: service/src/main/kotlin/no/nav/melosys/service/dokument/brev/BrevDataInnvilgelseFlereLand.kt
package no.nav.melosys.service.dokument.brev

import no.nav.melosys.domain.Lovvalgsperiode
import no.nav.melosys.domain.avklartefakta.AvklartVirksomhet
import no.nav.melosys.domain.kodeverk.Landkoder

class BrevDataInnvilgelseFlereLand(
    brevbestillingDto: BrevbestillingDto,
    saksbehandler: String
) : BrevData(brevbestillingDto, saksbehandler) {
    var arbeidsgivere: Collection<AvklartVirksomhet> = emptyList()
    var alleArbeidsland: List<String> = emptyList()
    var ukjenteEllerAlleEosLand: Boolean = false
    var lovvalgsperiode: Lovvalgsperiode? = null
    var avklartMaritimTypeSokkel: Boolean = false
    var avklartMaritimTypeSkip: Boolean = false
    var bostedsland: String? = null
    var begrensetPeriode: Boolean = false
    var marginaltArbeid: Boolean = false
    var trydemyndighetsland: Landkoder? = null

    var vedleggA1: BrevDataA1? = null
}
